DiverseSelector/methods/tests/test_partition.py

```diff
@@ -81,30 +82,118 @@ def test_directed_sphere_on_line_with_larger_radius():
     assert_equal(selector.r, 1.0)
 
 
-# def test_gridpartitioning():
-#     """Testing DirectedSphereExclusion class."""
-#     coords, _, _ = generate_synthetic_data(n_samples=100,
-#                                            n_features=2,
-#                                            n_clusters=1,
-#                                            pairwise_dist=True,
-#                                            metric="euclidean",
-#                                            random_state=42)
-
-#     coords_cluster, class_labels_cluster, _ = generate_synthetic_data(n_samples=100,
-#                                                                       n_features=2,
-#                                                                       n_clusters=3,
-#                                                                       pairwise_dist=True,
-#                                                                       metric="euclidean",
-#                                                                       random_state=42)
-#     selector = GridPartitioning(cells=3)
-#     selected_ids = selector.select(arr=coords_cluster, size=12, labels=class_labels_cluster)
-#     # make sure all the selected indices are the same with expectation
-#     assert_equal(selected_ids, [2, 25, 84, 56, 8, 70, 58, 78, 4, 46, 65, 29])
-
-#     selector = GridPartitioning(cells=3)
-#     selected_ids = selector.select(arr=coords, size=12)
-#     # make sure all the selected indices are the same with expectation
-#     assert_equal(selected_ids, [7, 55, 70, 57, 29, 91, 9, 65, 28, 11, 54, 88])
+@pytest.mark.parametrize("numb_pts", [4])
+@pytest.mark.parametrize("method", ["equifrequent", "equisized"])
+def test_grid_partitioning_independent_on_simple_example(numb_pts, method):
+    r"""Test grid partitioning on an example each bin has only one point."""
+    # Construct feature array where each molecule is known to be in which bin
+    # The grid is a uniform grid from 0 to 10 in X-axis and 0 to 11 on y-axis.
+    x = np.linspace(0, 3, numb_pts)
+    y = np.linspace(0, 3, numb_pts)
+    X, Y = np.meshgrid(x, y)
+    grid = np.array([1.0, 0.0]) + np.vstack([X.ravel(), Y.ravel()]).T
+    # Make one bin have an extra point
+    grid = np.vstack((grid, np.array([1.1, 0.0])))
+
+    # Here the number of cells should be equal to the number of points in each dimension
+    #  excluding the extra point, so that the answer is unique/known.
+    selector = GridPartitioning(numb_bins_axis=4, grid_method=f"{method}_independent")
+    # Sort the points so that they're comparable to the expected answer.
+    selected_ids = np.sort(selector.select(grid, size=len(grid) - 1))
+    expected = np.arange(len(grid) - 1)
+    assert_equal(selected_ids, expected)
+
+
+def test_grid_partitioning_equisized_dependent_on_simple_example():
```

Comment:
@Ali-Tehrani, this test fails for me. Did you cook up this test yourself or it was taken from a paper/code? I took a closer look, and it seems to be because all 9 bins have the same diversity, which does not seem correct. See:
```bash
[(142.15618366216015, (4, 4)), (142.15618366216015, (4, 3)), (142.15618366216015, (4, 0)), (142.15618366216015, (2, 4)), (142.15618366216015, (2, 0)), (142.15618366216015, (1, 4)), (142.15618366216015, (1, 0)), (142.15618366216015, (0, 4)), (142.15618366216015, (0, 0))]
```
This is most likely due to @FanwangM's rebasing. Because he has forced-pushed, I can't get the original test output. @Ali-Tehrani, can you please share your original branch with me?

@FanwangM, did all tests pass on your laptop before you force-pushed?

Reply:
Thanks for the fix @Ali-Tehrani.